packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -464,29 +474,41 @@ class HotRunner extends ResidentRunner {
     if (benchmarkMode)
       reassembleTimer.start();
     // Reload the isolate.
-    await currentView.uiIsolate.reload();
+    for (FlutterView view in currentViews) {
+      await view.uiIsolate.reload();
+    }
     // We are now running from source.
     _runningFromSnapshot = false;
     // Check if the isolate is paused.
-    final ServiceEvent pauseEvent = currentView.uiIsolate.pauseEvent;
-    if ((pauseEvent != null) && (pauseEvent.isPauseEvent)) {
-      // Isolate is paused. Stop here.
+
+    final List<FlutterView> reassembleViews = <FlutterView>[];
+    for (FlutterView view in currentViews) {
+      final ServiceEvent pauseEvent = view.uiIsolate.pauseEvent;
+      if ((pauseEvent != null) && (pauseEvent.isPauseEvent)) {
+        // Isolate is paused. Don't reassemble.
+        continue;
+      }
+      reassembleViews.add(view);
+    }
+    if (reassembleViews.isEmpty) {
       printTrace('Skipping reassemble because isolate is paused.');
       return new OperationResult(OperationResult.ok.code, reloadMessage);
     }
     await _evictDirtyAssets();
     printTrace('Reassembling application');
-    try {
-      await currentView.uiIsolate.flutterReassemble();
-    } catch (_) {
-      printError('Reassembling application failed.');
-      return new OperationResult(1, 'error reassembling application');
-    }
-    try {
-      /* ensure that a frame is scheduled */
-      await currentView.uiIsolate.uiWindowScheduleFrame();
-    } catch (_) {
-      /* ignore any errors */
+    for (FlutterView view in reassembleViews) {
+      try {
+        await view.uiIsolate.flutterReassemble();
+      } catch (_) {
```

Comment:
please don't catch exceptions unconditionally.
If you're expecting some exceptions here, list them explicitly.